multimodal/src/autogluon/multimodal/predictor.py

```diff
@@ -744,6 +853,16 @@ def fit(
         self.elapsed_time = (training_end - training_start) / 60.0
         return self
 
+    def _verify_inference_ready(self):
+        if not self._fit_called and not self._model_loaded:
```

Comment:
I guess the criteria of ready for inference may need to be reconsidered. If a the fit corrupted and we load the last checkpoint, is it ready for inference?

Reply:
The `self._fit_called` is set True at the beginning of `fit()`, but the predictor is not actually fit yet. It may cause some wrong check here in some special cases I guess.

Reply:
Previously there is no check of `inference_ready` so we are better by providing some error messages at least.

Reply:
Besides, I feel this logic is not straightforward as this is affected by several flags. May need to simply it.

Reply:
Basically, if the scenario can be detected by `_verify_inference_ready`, we will raise the RuntimeError. Otherwise, the behavior is the same as the previous implementation.

Reply:
We can mark it as a lower-priority TODO and fix it in a follow-up PR.